File: sdk/runanywhere-react-native/src/Data/Models/Downloading/DownloadTask.ts
/**
 * DownloadTask.ts
 *
 * Download task information
 *
 * Reference: sdk/runanywhere-swift/Sources/RunAnywhere/Data/Models/Downloading/DownloadTask.swift
 */

import type { DownloadProgress } from './DownloadProgress';

/**
 * Download task information
 */
export interface DownloadTask {
  readonly id: string;
  readonly modelId: string;
  readonly progress: AsyncGenerator<DownloadProgress, void, unknown>;
  readonly result: Promise<string>; // URL as string
}

File: sdk/runanywhere-flutter/lib/core/protocols/downloading/download_task.dart
import 'dart:async';

import 'download_progress.dart';

/// Download task information
/// Matches iOS DownloadTask from Data/Models/Downloading/DownloadTask.swift
class DownloadTask {
  /// Unique task identifier
  final String id;

  /// Model identifier being downloaded
  final String modelId;

  /// Stream of progress updates
  final Stream<DownloadProgress> progress;

  /// Future that completes with the download result (file URI)
  final Future<Uri> result;

  const DownloadTask({
    required this.id,
    required this.modelId,
    required this.progress,
    required this.result,
  });

  @override
  String toString() => 'DownloadTask(id: $id, modelId: $modelId)';
}
